api.js

```diff
@@ -1,116 +1,55 @@
-            forecastItemContent += `${result.temp2m.max},`;
-          }
-
-          if (result?.temp2m?.min) {
-            forecastItemContent += `${result.temp2m.min},`;
-          }
-          const dailyForecast = cityDailyForecast[forecastResults.indexOf(result)];
-
-          const dateResult = forecastItemContent.substring(4, forecastItemContent.indexOf(","));
-          forecastItemContent = forecastItemContent.substring(forecastItemContent.indexOf(",") + 1, forecastItemContent.length);
-          const weatherResult = forecastItemContent.substring(0, forecastItemContent.indexOf(","));
-          forecastItemContent = forecastItemContent.substring(forecastItemContent.indexOf(",") + 1, forecastItemContent.length);
-          const maxTempResult = forecastItemContent.substring(0, forecastItemContent.indexOf(","));
-          forecastItemContent = forecastItemContent.substring(forecastItemContent.indexOf(",") + 1, forecastItemContent.length);
-          const minTempResult = forecastItemContent.substring(0, forecastItemContent.indexOf(","));
-          const forecastItemsList = dailyForecast.getElementsByClassName("forecast-item-info");
-
-          for (let i = 0; i < forecastItemsList.length; i++) {
-            const forecastItemInfo = forecastItemsList[i];
-            if (i === 0) {
-              if (getCurrentDate() === dateResult) {
-                forecastItemInfo.innerText = "Today";
-              } else {
-                const formattedDate = `${dateResult.substring(2, 4)} ${format(dateResult.substring(0, 2), "MMM")}`;
-                forecastItemInfo.innerText = formattedDate;
-                // console.log(formattedDate);
-              }
-            } else if (i === 1) {
-              forecastItemInfo.innerHTML = weatherIconConverter(weatherResult);
-            } else if (i === 2) {
-              forecastItemInfo.innerText = `${minTempResult} °C / ${maxTempResult} °C`;
-            }
-          }
+      return response.json();
+    })
+    .then((jsonResponse) => {
+      try {
+        let jsonValue = jsonResponse;
+        if (
+          typeof jsonResponse === "string"
+          || jsonResponse instanceof String
+        ) {
+          jsonValue = JSON.parse(jsonResponse);
         }
-
-        // Showing the forecast after everything has been loaded.
-        if (location === "custom") {
-          document.getElementById("popup-loading-container").style.visibility = "hiddem";
-          document.getElementById("popup-loading-container").style.display = "none";
-          document.getElementById("popup-forecast-container").style.visibility = "visible";
-          document.getElementById("popup-forecast-container").style.display = "flex";
+        // Validating response
+        if (!jsonValue?.dataseries) {
+          throw new Error(`Invalid response from the API: "${jsonValue}"`);
+        }
+        const forecastResults = jsonValue.dataseries;
+        console.log(forecastResults);
```

Comment:
Please don't leave `console.log` in your code. This is for debugging purposes and should not be available to the user.

Reply:
Thank you, I have removed all console logs from this file.